source/common/network/utility.cc

```diff
@@ -133,6 +133,23 @@ Utility::parseInternetAddressAndPort(const std::string& ip_address) {
   return std::make_shared<Address::Ipv4Instance>(&sa4);
 }
 
+Address::InstanceConstSharedPtr
+Utility::fromProtoResolvedAddress(const envoy::api::v2::ResolvedAddress& resolved_address) {
+  switch (resolved_address.address_case()) {
+  case envoy::api::v2::ResolvedAddress::kSocketAddress:
+    // TODO(htuch): Can do this more efficiently if it matters with direct sockaddr manipulation,
+    // keeping it simple for now.
+    return parseInternetAddressAndPort(
+        fmt::format("{}:{}", resolved_address.socket_address().ip_address(),
+                    resolved_address.socket_address().port().value()));
+  case envoy::api::v2::ResolvedAddress::kPipe:
+    return Address::InstanceConstSharedPtr{
+        new Address::PipeInstance(resolved_address.pipe().path())};
+  default:
```

Comment:
I think this has come up before. The proto `oneof` case enum includes the "not set" case, which needs to be handled in the switch, or you get:

```
source/common/network/utility.cc: In static member function 'static Envoy::Network::Address::InstanceConstSharedPtr Envoy::Network::Utility::fromProtoResolvedAddress(const envoy::api::v2::ResolvedAddress&)':
source/common/network/utility.cc:138:10: error: enumeration value 'ADDRESS_NOT_SET' not handled in switch [-Werror=switch]
   switch (resolved_address.address_case()) {
```

In general, BTW, `oneof`is basically a switch on an integer rather than typed `enum class`, as the cases are basically arbitrary protobuf field numbers.